src/App.tsx

```diff
@@ -83,6 +85,67 @@ export default class App extends Component<{
         });
     }
 
+    selectParentNode = () =>
+    {
+        const root = this.state.ourTree.root
+        const node = this.state.ourTree._selectNodeById(this.state.chosenNode)
+        if (node !== root)
+        {
+            const parent = root.getParent(node, root)
+        
+            this.setState({
+                chosenNode: parent.id
+            })
+        }
+    }
+
+    selectChildNode = () =>
+    {
+        const node = this.state.ourTree._selectNodeById(this.state.chosenNode)
+        if (node.children.length !== 0)
+        {
+            const child = node.children[0]
+        
+            this.setState({
+                chosenNode: child.id
+            })
+        }
+    }
+
+    selectNextChildNode = () =>
```

Comment:
Can I somehow simplify these two methods?

Reply:
Got a few ideas, but let's extract the component first